feat: adds ability to serve a digest email

as1729 opened the pull request:
### Ticket #19 

### Description
This PR does a few things: 
1. Adds the grants digest email
2. Re-uses components across multiple emails
3. Ensures all emails have consistent styling
4. Ensures there's a placeholder for the notification management changes in the front-end

### Screenshots / Demo Video
Demoed during show and tell
https://www.notion.so/usdr/Show-Tell-d4ba5e125d2b49df844eda046d298fd3

### Testing
- Added unit tests and integration tests

### Checklist
- [x] Provided ticket and description
- [x] Your PR title contains the ISSUE ticket number e.g "86: ..."
- [x] Provided screenshots/demo
- [x] Provided testing information
- [x] Provided adequate test coverage for all new code
- [x] Run automated tests (docker compose exec app yarn test)
- [x] Added PR reviewers
- [ ] Ensure at least 1 review before merging

as1729 commented on packages/server/src/lib/email.js:
```diff
@@ -123,11 +186,74 @@ async function sendGrantAssignedEmail({ grantId, agencyIds, userId }) {
 }
 
+async function sendGrantDigestForAgency(agency) {
+    const newGrants = await db.getNewGrantsForAgency(agency);
```
Add some additional logging here for important events.

as1729 commented on packages/server/src/lib/email.js:
```diff
@@ -123,11 +186,74 @@ async function sendGrantAssignedEmail({ grantId, agencyIds, userId }) {
+    }
+
+    const grantDetails = [];
+    newGrants.slice(0, 3).forEach((grant) => grantDetails.push(module.exports.getGrantDetail(grant)));
```
Check with knex if it can get total count while getting the first 3 results only

as1729 replied:
make some screenshots of metrics dashboard in render to get an idea for typical usage and memory usage.

as1729 commented on packages/server/src/lib/email.js:
```diff
@@ -123,11 +186,74 @@ async function sendGrantAssignedEmail({ grantId, agencyIds, userId }) {
+        title: 'New Grants Digest',
+        notifications_url: 'mailto:grants-helpdesk@usdigitalresponse.org?subject=Unsubscribe&body=Please unsubscribe me from the grant digest notification email.',
+    });
+    const emailPlain = emailHTML;
```
Just leave this empty instead.

as1729 replied:
Done

TylerHendrickson reviewed (changes requested):
(Changes discussed in call)

as1729 commented on packages/server/src/db/index.js:
```diff
@@ -284,6 +285,20 @@ function deleteKeyword(id) {
+    const agencyCriteria = await getAgencyCriteriaForAgency(agency.id);
+
+    const rows = await knex(TABLES.grants)
+        .select(knex.raw(`${TABLES.grants}.*, count(*) OVER() AS total_grants`))
```
@TylerHendrickson thanks for pushing on this! I was able to ensure that we only returned at most 3 grants while also getting a count of the rows directly from postgres in a single query.

I implemented it as a window-function and will take a next step once we are using the feature to analyze the query a bit more to see how postgres is processing this.

as1729 commented:
@TylerHendrickson I added 3 main changes:
- Simple plaintext parser using regex. This isn't great and I added a TODO to revisit this in the next PR. But for now its trending in the right direction.
- Added important event logs
- Ensured that we retrieve grant totals in a single query so we aren't overloading application memory with unnecessary grants data.

TylerHendrickson commented on packages/server/src/lib/email.js:
```diff
@@ -159,7 +160,9 @@ async function sendGrantAssignedNotficationForAgency(assignee_agency, grantDetai
-    const emailPlain = emailHTML;
+
+    // TODO: add plain text version of the email
+    const emailPlain = emailHTML.replace(/<[^>]+>/g, '');
```
This is a clever solution for the interim. I'd replace the matches here with a single space so that words/sentences that are separated by newlines aren't smooshed together.

Example:
```
Some information is here.
And this is also useful.
```
Should output as `Some information is here. And this is also useful.` rather than `Some information is here.And this is also useful.`

TylerHendrickson commented on packages/server/src/lib/email.js:
```diff
@@ -187,29 +190,37 @@ async function sendGrantAssignedEmail({ grantId, agencyIds, userId }) {
     const formattedBody = mustache.render(formattedBodyTemplate.toString(), {
         body_title: 'New grants have been posted',
-        body_detail: `There are ${newGrants.length} new grants matching your agency's keywords and settings.`,
+        body_detail: `There are ${newGrants[0].total_grants} new grants matching your agency's keywords and settings.`,
```
IMO it's fine for now (I think people are generally used to this type of "default plural" behavior), but as we add more and more emails, it would be nice to have some kind of conventional/parameterized way of conditionally formatting plural/singular words so that we can avoid errant plurals in cases when the content is singular (e.g. "There is 1 new grant matching..." instead of "There are 1 new grants matching..."). 

Perhaps mustache has something akin to Django's [pluralize filter](https://docs.djangoproject.com/en/4.1/ref/templates/builtins/#pluralize)?

TylerHendrickson reviewed (approved):
@as1729 Looks good to me! I added a few notes for your consideration, but nothing I think is a show-stopper. 

Great work putting this together! 👏 👏 👏
